core/server/proxy/src/main/java/alluxio/proxy/s3/S3RestUtils.java

```diff
@@ -464,6 +464,20 @@ public static void setEntityTag(FileSystem fs, AlluxioURI objectUri, String enti
         .build());
   }
 
+  /**
+   * This helper method is used to get the ETag xAttr on an object.
+   * @param status The {@link URIStatus} of the object
+   * @return the entityTag String, or null if it does not exist
+   */
+  public static String getEntityTag(URIStatus status) {
+    if (status.getXAttr() == null
+        || !status.getXAttr().containsKey(S3Constants.ETAG_XATTR_KEY)) {
+      return null;
```

Comment:
I debated whether or not we should compute the ETag at this point instead of returning `null`, but decided against it since it would incur a lot of network overhead (2 round-trips). If/when ETags become more completely supported, we may want to revisit that.

Reply:
I think so. maybe the better way to solve it is to compute the ETag when creating it or get the ETag attribute of the object from UFS if UFS supports it.

Reply:
There's no precedent for enforcing Inodes to compute checksums so the only decision we have is whether or not we want to do so at the Alluxio Proxy level.
- Fetching from UFS would require unique code for each UFS's API and be harder to maintain; unless we enforce it as part of an Alluxio interface (which we're not trying to do)

We _do_ create the ETag when the object is created, but this is about addressing objects in Alluxio which were not created through the Alluxio S3 API.